src/bin/ion/commands/mod.rs

```diff
@@ -17,6 +25,14 @@ pub trait IonCliCommand {
     /// A brief message describing this command's functionality.
     fn about(&self) -> &'static str;
 
+    /// If `true`, prevents this command from showing up in the parent namespace's help message.
+    /// This can be helpful for deprecating names while still supporting the old alias
+    ///
+    /// Defaults to `false`.
+    fn hide_from_help_message(&self) -> bool {
+        false
+    }
```

Comment:
🗺️ This features allows `dump` to be omitted from `ion --help`'s output.

![image](https://github.com/amazon-ion/ion-cli/assets/611616/064786b4-afe9-43e6-aba7-64233756f34a)

But usages of the command are still accepted/recognized:

![image](https://github.com/amazon-ion/ion-cli/assets/611616/0a16ceee-45ae-41bc-b16f-ce5261247cfa)